grpc/server.go

```diff
@@ -129,6 +129,15 @@ func (sb *serverBuilder) Build(tlsConfig *tls.Config, statsRegistry prometheus.R
 		}
 	}
 
+	// Ensure that the health service has the same ClientNames as the other
+	// services, so that health checks can be performed by clients which are
+	// allowed to connect to the server.
+	healthService := sb.cfg.Services[healthpb.Health_ServiceDesc.ServiceName]
+	for as := range acceptedSANs {
+		healthService.ClientNames = append(healthService.ClientNames, as)
+	}
+	sb.cfg.Services[healthpb.Health_ServiceDesc.ServiceName] = healthService
```

Comment:
I believe this can be condensed down to merely
```suggestion
	sb.cfg.Services[healthpb.Health_ServiceDesc.ServiceName].ClientNames = acceptedSANs
```
because any client names which were specifically configured for the health service will have already been incorporated into acceptedSANs by the loop above.